### ISSUE TITLE
SNMP Agent - Responding with Error Status?

### ISSUE THREAD

**Issue by morrisrob** [markabrahams/node-net-snmp#140]
I'm using your Command Responder / SNMP agent and I have a question about responding with an error response.  For example, if a user submits an SNMP requests for a valid object, but we're unable to provide a valid response due to any number of reasons in our application (ie. we can't communicate with a third party service), I'd like to provide a genError to the user.  I've looked through your documentation and am trying to determine if this feature is supported.  I see how I can react to these types of messages when sending a command with your Command Generator tool, but can't find if I'm able to generate these as an agent.  Thank you.

**Comment by markabrahams** [comment]
Hi @morrisrob - this feature is not currently supported, but I'll keep this open for implementation, since it would a useful option for the agent to have.

**Comment by markabrahams** [comment]
Hi @morrisrob - I've implemented this now and published in version 3.0.3 of the npm.

You can now pass an optional error object to the done() function in the handler, where the error object has an "errorStatus" key that can take any value from the snmp.ErrorStatus constant e.g. for a somewhat contrived handler function that always returns a genError, use:

```
function (mibRequest) {
   mibRequest.done ({
       errorStatus: snmp.ErrorStatus.GeneralError
   });
}
```
